Composer/packages/client/src/pages/botProject/skill-configuration/IsSkill.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core';
+import React from 'react';
+import { useRecoilValue } from 'recoil';
+import { Toggle } from 'office-ui-fabric-react/lib/components/Toggle';
+import cloneDeep from 'lodash/cloneDeep';
+import formatMessage from 'format-message';
+
+import { dispatcherState, rootBotProjectIdSelector, settingsState } from '../../../recoilModel';
+import { mergePropertiesManagedByRootBot } from '../../../recoilModel/dispatchers/utils/project';
+
+export const toggle = css`
+  display: flex;
+  flex-direction: column;
+  box-sizing: border-box;
+`;
+
+type Props = {
+  projectId: string;
+};
+
+export const IsSkill: React.FC<Props> = ({ projectId }) => {
+  const { setSettings } = useRecoilValue(dispatcherState);
+  const rootBotProjectId = useRecoilValue(rootBotProjectIdSelector) || '';
+  const settings = useRecoilValue(settingsState(projectId));
+  const mergedSettings = mergePropertiesManagedByRootBot(projectId, rootBotProjectId, settings);
+  const { skillConfiguration } = mergedSettings;
+
+  const setIsSkill = React.useCallback(
+    (isSkill: boolean) => {
+      const updatedSetting = {
+        ...cloneDeep(mergedSettings),
+        skillConfiguration: { ...skillConfiguration, isSkill },
+      };
+      setSettings(projectId, updatedSetting);
+    },
+    [mergedSettings, projectId, skillConfiguration]
+  );
+
+  const toggleIsSKill = React.useCallback(
+    (event, checked?: boolean) => {
+      setIsSkill(!!checked);
+    },
+    [setIsSkill]
```

Comment:
Isn't it required since `isSetSkill` can update?

```
const setIsSkill = React.useCallback(
  (isSkill: boolean) => {
    const updatedSetting = {
      ...cloneDeep(mergedSettings),
      skillConfiguration: { ...skillConfiguration, isSkill },
    };
    setSettings(projectId, updatedSetting);
  },
  [mergedSettings, projectId, skillConfiguration]
);

const toggleIsSKill = React.useCallback(
  (event, checked?: boolean) => {
    setIsSkill(!!checked);
  },
  [setIsSkill]
);

```

Reply:
maybe give it a better name, I read it as if it's a state setter. ```updateSettings```?

Reply:
Yea, I’m following the same pattern for the other bot setting sections, so I don’t want to deviate from them too much. I agree, `set*’` does sound like a react state.